aether-kernel/aether/kernel/api/migrations/0103_submission_no_cascade_delete.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 2.2.1 on 2019-05-08 15:17
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def migrate__update_submissions(apps, schema_editor):
+    Submissions = apps.get_model('kernel', 'Submission')
+    for submission in Submissions.objects.all():
+        submission.save() # ensures project from mappingset is linked to submission
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('kernel', '0102_schema_decorator'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='submission',
+            name='mappingset',
+            field=models.ForeignKey(
+                null=True,
+                on_delete=django.db.models.deletion.SET_NULL,
+                related_name='submissions',
+                to='kernel.MappingSet',
+                verbose_name='mapping set',
+            ),
+        ),
+        migrations.AlterField(
+            model_name='submission',
+            name='project',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='submissions', to='kernel.Project', verbose_name='project'),
+        ),
+        migrations.RunPython(
```

Comment:
shouldn't this migration be before the `submission.project` one? I mean, please ensure that all the submission entries have `project` field and after that alter the field.